js_modules/dagster-ui/packages/ui-core/src/assets/Stale.tsx

```diff
@@ -127,50 +175,158 @@ function groupedCauses(
+              <StaleReason
+                key={idx}
+                reason={cause.reason}
+                dependency={cause.dependency}
+                assetKey={assetKey}
+              />
+            ))}
+          </Box>
+        );
+      })}
     </Box>
-  </Box>
-);
+  );
+};
 
 const StaleReason = ({
   reason,
   dependency,
 }: {
+  assetKey: AssetKeyInput;
   reason: string;
   dependency: AssetNodeKeyFragment | null;
 }) => {
-  if (!dependency) {
-    return <Caption>{` ${reason}`}</Caption>;
-  }
+  const content = useMemo(() => {
+    if (!dependency) {
+      return <Caption>{` ${reason}`}</Caption>;
+    }
+
+    const dependencyName = displayNameForAssetKey(dependency);
+    const dependencyPythonName = dependencyName.replace(/ /g, '');
+    if (reason.endsWith(`${dependencyPythonName}`)) {
+      const reasonUpToDep = reason.slice(0, -dependencyPythonName.length);
+      return (
+        <>
+          <Caption>{reasonUpToDep}</Caption>
+          <Link to={assetDetailsPathForKey(dependency)}>
+            <Tag icon="asset">{dependencyName}</Tag>
+          </Link>
+        </>
+      );
+    }
 
-  const dependencyName = displayNameForAssetKey(dependency);
-  const dependencyPythonName = dependencyName.replace(/ /g, '');
-  if (reason.endsWith(`${dependencyPythonName}`)) {
-    const reasonUpToDep = reason.slice(0, -dependencyPythonName.length);
     return (
-      <Caption>
-        {` ${reasonUpToDep}`}
-        <Link to={assetDetailsPathForKey(dependency)}>{dependencyName}</Link>
-      </Caption>
+      <>
+        <Link to={assetDetailsPathForKey(dependency)}>
+          <Tag icon="asset">{dependencyName}</Tag>
+        </Link>
+        <Caption>{` ${reason} `}</Caption>
+      </>
     );
-  }
+  }, [dependency, reason]);
+  return (
+    <Box
+      padding={{vertical: 8, horizontal: 12}}
+      flex={{direction: 'row', alignItems: 'center', gap: 4}}
+    >
+      {content}
+    </Box>
+  );
+};
 
+export const NewInBranchTag = ({
+  changedReasons,
+  assetKey,
+}: {
+  changedReasons: ChangeReason[];
+  assetKey: AssetKeyInput;
+}) => {
+  const changes = changedReasons.filter((reason) => reason !== ChangeReason.NEW);
+
+  function getDescription(change: ChangeReason) {
+    switch (change) {
+      case ChangeReason.NEW:
+        return '';
+      case ChangeReason.CODE_VERSION:
+        return 'has a modified code version';
+      case ChangeReason.INPUTS:
+        return 'has modified dependencies';
+      case ChangeReason.PARTITIONS_DEFINITION:
+        return 'has a modified partition definition';
+    }
+  }
   return (
-    <Caption>
-      {` ${reason} `}(<Link to={assetDetailsPathForKey(dependency)}>{dependencyName}</Link>)
-    </Caption>
+    <Popover
+      position="top"
+      content={
+        <Box flex={{direction: 'column'}}>
+          <Box padding={{horizontal: 12, vertical: 8}} border="bottom">
+            <Subtitle2>
+              {numberFormatter.format(changes.length)}{' '}
```

Comment:
minor: is it possible to disable this popover if the asset is new? in that case it won't have any changes wrt to the parent deployment since it's a brand new asset, but the "0 changes" popover sort of implies that there could be changes

<img width="378" alt="Screenshot 2024-03-06 at 9 39 26 AM" src="https://github.com/dagster-io/dagster/assets/19783112/8d5c2b7b-82b5-44a8-9b14-7d080b46174a">

Reply:
done